### PR TITLE
Dev 3736 improvement/xss protection

### PR DESCRIPTION
**Pull request by Achalhii**
In this PR, we add an HTML-sanitizer in post/put/patch docs and versioned-docs endpoint to prevent XSS injection.

Do to this, we use https://www.npmjs.com/package/sanitize-html library.

I tested `sanitizeHTMLFromXSS()` by applying it on the GET endpoints of our api, then I did several tests on the BO.

### CODE REVIEW

File: services/auth/lib/handlers.js [hunk 1]
```diff
@@ -588,6 +589,28 @@ function isEmailValid(email) {
...
 }
 
+/**
+ * The function `sanitizeHTMLFromXSS` is used to sanitize HTML strings and objects from potential cross-site scripting (XSS) attacks.
+ * @param obj - The `obj` parameter is the input object that you want to sanitize from potential XSS (Cross-Site Scripting) attacks.
+ * It can be a string or an object containing strings.
+ * @returns The function `sanitizeHTMLFromXSS` returns the sanitized version of the input object, with any potentially harmful HTML
+ * tags or attributes removed.
+ */
+function sanitizeHTMLFromXSS(obj) {
```
Comment (gabila-r) [inline services/auth/lib/handlers.js:599]: Why did you put this function inside the auth service ? 🤔 It has nothing to do in auth service for me !

**Review by gabila-r — CHANGES_REQUESTED** [review]

File: services/auth/lib/handlers.js (hunk 1 above)
Comment (Achalhii) [reply services/auth/lib/handlers.js:599]: My bad, I was looking for a "utils" file to put it in, but I couldn't find anything, so I fell back on this one.

Do you have a better suggestion ? Create a utils.js file ?

Comment (Achalhii) [reply services/auth/lib/handlers.js]: Ok, I find a good solution

**Review by gabila-r — APPROVED** [review]

File: lib/modules/sanitize.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ * @returns The function `sanitizeHTMLFromXSS` returns the sanitized version of the input object,
+ * with any potentially harmful HTML tags or attributes removed.
+ */
+function sanitizeHTMLFromXSS(obj) {
+  if (!obj) return obj;
+  switch (typeof obj) {
+    case 'string':
+      return sanitizeHtml(obj);
+    case 'object':
+      return Object.entries(obj).reduce((acc, [key, value]) => {
```
Comment (rombat) [inline lib/modules/sanitize.js:17]: beware that a Date has an object type.
but `Object.entries()` on a Date returns an empty array. I don't think you want that.
Don't know if we ever have the case where it's used, but you should handle it imho. I've already had an issue with this case. Same for ObjectId btw.
I did something like that:
`const exclude = Object.prototype.toString.call(value) === '[object Date]' || ObjectId.isValid(value);`

**Review by rombat — CHANGES_REQUESTED** [review]

File: lib/modules/sanitize.js (hunk 2 above)
Comment (Achalhii) [reply lib/modules/sanitize.js:17]: I already meet this problem during my test and i fixed it. Look the `initialValue` of the reduce 😉.

Comment (rombat) [reply lib/modules/sanitize.js:17]: nice, I wasn't aware of that ! 👍🏼

**Review by rombat — APPROVED** [review]

**Pull request merged**